### PR TITLE
feat:add refresh token

### PR DESCRIPTION
**Pull request by Podgore**

### CODE REVIEW

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 1]
```diff
@@ -56,30 +70,105 @@ public async Task<AuthSuccessDTO> RegisterAsync(RegisterUserDTO user)
         if (!result.Succeeded)
             throw new UserManagerException($"User manager operation failed:\n", result.Errors);
 
-        return new AuthSuccessDTO(GenerateJwtToken(newUser));
+        return new AuthSuccessDTO(GenerateJwtToken(newUser), GenerateRefreshTokenAsync(newUser));
+    }
+
+    public async Task<AuthSuccessDTO> RefreshTokenAsync(string accessToken, string refreshToken)
+    {
+
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:78]: empty line should be removed, I guess

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 2]
```diff
@@ -56,30 +70,105 @@ public async Task<AuthSuccessDTO> RegisterAsync(RegisterUserDTO user)
...
+    {
+        var jwtTokenHandler = new JwtSecurityTokenHandler();
+        var validationParametrs = _tokenValidationParametrs.Clone();
+        validationParametrs.ValidateLifetime = false;
+        try
+        {
+            var principal = jwtTokenHandler.ValidateToken(token, validationParametrs, out var validatedToken);
+
+            if (!IsJwtWithValidSecurityAlgorithm(validatedToken))
+            {
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:115]: remove curly brackets for this if

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 3]
```diff
@@ -56,30 +70,105 @@ public async Task<AuthSuccessDTO> RegisterAsync(RegisterUserDTO user)
...
+        }
+        catch
+        {
+            throw new TokenValidatorException("Something went wrong with token validator");
+        }
+    }
+
+    private bool IsJwtWithValidSecurityAlgorithm(SecurityToken validatedToken)
+    {
+        return validatedToken is JwtSecurityToken jwtSecurityToken &&
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:129]: fix formatting

File: StudyHub/StudyHub.Common/DTO/AuthDTO/RefreshTokenRequest.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace StudyHub.Common.DTO.AuthDTO;
+public class RefreshTokenRequest
+{
+    public string RefreshToken { get; set; } = string.Empty;
+    public string Token { get; set; } = string.Empty;
+
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/DTO/AuthDTO/RefreshTokenRequest.cs:6]: empty line

File: StudyHub/StudyHub.Common/Exceptions/ExpiredError.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace StudyHub.Common.Exceptions;
+public class ExpiredError : Exception
+{
+    public ExpiredError
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/Exceptions/ExpiredError.cs:10]: formatting

File: StudyHub/StudyHub.Common/Exceptions/IncorrectParametersError.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace StudyHub.Common.Exceptions;
+public class IncorrectParametersError : Exception
+{
+    public IncorrectParametersError
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/Exceptions/IncorrectParametersError.cs:10]: also formatting

File: StudyHub/StudyHub.Common/Exceptions/InvalidSecurityAlgorithmException.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace StudyHub.Common.Exceptions;
+public class InvalidSecurityAlgorithmException : Exception
+{
+    public InvalidSecurityAlgorithmException
```
Comment (aquaprogit) [inline StudyHub/StudyHub.Common/Exceptions/InvalidSecurityAlgorithmException.cs:10]: same and for others Exceptions

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by fedyafedko — APPROVED** [review]

**Review by aquaprogit — APPROVED** [review]

**Review by aquaprogit — CHANGES_REQUESTED** [review]
rename "Errors" in StudyHub.Common.**Exceptions**

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 8]
```diff
@@ -82,4 +149,20 @@ private string GenerateJwtToken(User user)
...
         return jwtToken;
     }
+    private string GenerateRefreshTokenAsync(User user)
+    {
+        var refreshToken = new RefreshToken
+        {
+            UserId = user.Id,
+            Token = Convert.ToBase64String(RandomNumberGenerator.GetBytes(64)),
+            CreationDate = DateTime.UtcNow,
+            ExpiryDate = DateTime.Now.AddDays(7),
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:159]: better to move this to config

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 9]
```diff
@@ -32,10 +42,10 @@ public async Task<AuthSuccessDTO> LoginAsync(LoginUserDTO user)
 
         var result = await _userManager.CheckPasswordAsync(existingUser, user.Password);
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:43]: name it as "isPasswordValid"
use appropriate naming for variables

**Review by aquaprogit — CHANGES_REQUESTED** [review]

**Review by aquaprogit — APPROVED** [review]

**Pull request merged**